src/Boundary/AdminMapBoundary.java

```diff
@@ -169,4 +177,15 @@ public void moveNode(MapNode n, Point2D movedTo)
             edge.updateCost();
         }
     }
+
+    public void setCurrentKiosk (Kiosk kiosk) {
+        if (Hospital.kiosks.contains(kiosk)) {
+            Hospital.currentKiosk = kiosk;
+        } else {
```

Comment:
Is it possible to make kiosks protected and instead have a "getKiosks()" function or "containsKiosk" function